### PR TITLE
Add state msgs

### LINKED ISSUE #29: Abstract class for keeping track of field items
# 🚀 Feature Request
We want some way to represent the status of game pieces and different objects on the field, along with a certainty of them being in that state.

### LINKED ISSUE #80: Enumerate BAYs on the field
# 🚀 Feature Request

We need to establish a consistent standard to use across our system to describe the field. Some of the elements of the field have intrinsic, intuitive descriptors (ie. left vs right human player station, left vs right level 2 HAB, left vs right rocket, etc). However, there is no immediately intuitive way of enumerating BAYs on the field, and the manual doesn't give us a way.

Reminder that a BAY is the combination of a HATCH (the location that a HATCH PANEL can be placed) and its corresponding CARGO.

This is an open discussion - Please give feedback or alternatives to these formats. The first step is to pick an array format. Once we have that, we can determine how to enumerate the bays within each array.

### ~~:one: Unique ID for every bay on each half-field~~
 - ~~Represented as an array of 20 entries, convering a half-field. We would create two of these messages/datatypes/whatever, one for PARTNER_ALLIANCE and one for OPPOSING ALLIANCE~~
 - ~~I like separating the two alliances into separate objects because we use the data very differently, and frankly we care a lot less about the state of the opposing alliance's half field~~
 - ~~I like separating the two alliances into separate objects because it makes it easy to know which side of the field is ours, which side is theirs, without needing to parse a big array or something~~
image

### :two: Unique ID for every bay *per ship* on each half-field
 - Represented as three arrays (`left_rocket[6]`, `cargo_ship[8]`, `right_rocket[6]`).
 - Same as :one: with respect to two instances of the structure, one for each alliance
image

### ~~:three: Unique ID for every bay on the whole field~~
 - ~~Represented as one big array of 40 entries.~~
 - ~~Maybe 0-20 is always our alliance? Maybe 0-20 is always red? idk~~
image (1)

### ~~:four: Unique ID for every bay *per ship* on the whole field~~
 - ~~Represented as six arrays (red_left_rocket[6], blue_cargo_ship[8], etc)~~
image

## Enumeration within each array

Now that we know how the bays will be separate into arrays, we need to figure out how the bays within each array will be enumerated.

### Rockets
Within each rocket, I will advocate very strongly for bays on the same level to be sequential (ie. bottom-left is 0, bottom-right is 1). From a coding perspective, this makes it simple and intuitive to group a pair of bays at the same level, which will need to be done frequently. The calculation to convert bay to level is `floor(0/2) == floor(1/2) == 0`, `floor(2/2) == floor(3/2) == 1`, etc. I also think it makes sense for bays that are physically lower to have lower numbers. Therefore I think the following is the best option for enumeration:
Rocket

Feel free to disagree with the above, just please explain why and what you think is better.
If we do go with this, then the question is are even numbers:
 - :one: Always on the left when facing the rocket, as in the image above. (Even on right doesn't make sense, we read left to right)
 - :two: Always on the side of the rocket closest to the driver station
 - :three: Always on the side of the rocket furthest from the driver station

### Cargo ship
I think trying to pair opposite bays on the cargo ship is confusing and pointless. It makes sense on the rocket, where the pairings would be logical and have function, but I think the same scheme would be just be confusing and wouldn't have any purpose on the cargo ship. In my eyes, the only two logical options are to count clockwise or counter-clockwise around the cargo ship. I think CCW from above is prefereable since it's similar to left-to-right. Please let me know if there's a better way to enumerate these.
Cargo Ship
 - :one: Clockwise, from above (0 on the right of the field)
 - :two: Counter-clockwise, from above (0 on the left of the field, as shown in image)

### PR DESCRIPTION
**Pull request by Matthew-Reynolds**
# Pull Request

Closes #29 
Closes #80 

Adds messages that will be used throughout the stack for tracking the state of the field and of entities.

## Contribution Checklist

- [x] I have read the [contributing](https://github.com/uwreact/uwreact_robot/blob/master/CONTRIBUTING.md) guide.
- [x] I have referenced relevant issues.
- [x] I have labeled accordingly.
- [x] I have detailed my changes as much as possible.

## Change Checklist

- [ ] I have added tests.
- [x] I have added necessary documentation.
- [x] I have auto formatted using clang-format or yapf.
- [ ] I have tested my changes on real hardware.

### CODE REVIEW

File: uwreact_msgs/msg/EntityArray.msg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Header          header
```
Comment (wmmc88) [inline uwreact_msgs/msg/EntityArray.msg]: What's with the spacing in this file

File: uwreact_msgs/msg/RobotEntity.msg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+uint8 ALLIANCE_UNKNOWN = 0
+uint8 ALLIANCE_OPPOSING = 1
+uint8 ALLIANCE_PARTNER = 2
+
+Header        header
+GenericEntity entity
+uint8         alliance
+uint8         team_number
+float32       has_cargo_confidenceasdfasdfa
```
Comment (wmmc88) [inline uwreact_msgs/msg/RobotEntity.msg]: Wtf is this name

File: uwreact_msgs/msg/FieldState.msg [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+uint8 BAY_CARGO_SHIP_MID_LEFT    = 1
+uint8 BAY_CARGO_SHIP_NEAR_LEFT   = 2
+uint8 BAY_CARGO_SHIP_FRONT_LEFT  = 3
+uint8 BAY_CARGO_SHIP_FRONT_RIGHT = 4
+uint8 BAY_CARGO_SHIP_NEAR_RIGHT  = 5
+uint8 BAY_CARGO_SHIP_MID_RIGHT   = 6
+uint8 BAY_CARGO_SHIP_FAR_RIGHT   = 7
+
+Header    header
+float32   left_hp_cargo_confidence
```
Comment (wmmc88) [inline uwreact_msgs/msg/FieldState.msg]: Appending confidence in every name is overly verbose imo

File: uwreact_msgs/msg/RobotEntity.msg (hunk 2 above)
Comment (Matthew-Reynolds) [reply uwreact_msgs/msg/RobotEntity.msg]: Smh it was part of an experiment I was doing and I forgot to remove it. Thanks for pointing it out, fixed.

File: uwreact_msgs/msg/EntityArray.msg (hunk 1 above)
Comment (Matthew-Reynolds) [reply uwreact_msgs/msg/EntityArray.msg]: Fixed

File: uwreact_msgs/msg/FieldState.msg (hunk 3 above)
Comment (Matthew-Reynolds) [reply uwreact_msgs/msg/FieldState.msg]: Fair enough. Removed `_confidence`

**Comment by Matthew-Reynolds** [comment]
Why will float64s be used as inputs to the decay function, do we really need a full float64 for the confidences?

**Comment by wmmc88** [comment]
Upon receiving new information about state, the new confidence value will be a function of the old value, new value and confidence in the new value

**Comment by wmmc88** [comment]
Actually on second thought that has nothing to do with the msg and will be handled internally within the node itself

**Pull request merged**